Refactor ConnectorIdentity

luohao opened the pull request:
@dain @arhimondr

kokosing commented on presto-spi/src/main/java/io/prestosql/spi/security/ConnectorIdentity.java:
```diff
@@ -14,7 +14,6 @@
 package io.prestosql.spi.security;
 
 import java.security.Principal;
-import java.util.Objects;
```
please squash this with the previous commit that removed the usage of this import

kokosing commented on presto-spi/src/main/java/io/prestosql/spi/security/ConnectorIdentity.java:
```diff
@@ -47,32 +47,13 @@ public String getUser()
     }
 
-    @Override
-    public boolean equals(Object o)
```
You could mention in the commit what was refactored. Like:
```
Refactor ConnectorIdentity

- Remove not needed and incorrectly implemented hashCode and equals methods
- Return information about role from toString method.
```

findepi replied:
`ConnectorIdentity` equality might be used as Map keys. In fact, it _is_ used in our internal code.
Let's talk. @luohao @kokosing

kokosing replied:
I would merge it as it is, if it is related to our internal code we could handle that there.

findepi commented on presto-spi/src/main/java/io/prestosql/spi/security/ConnectorIdentity.java:
```diff
@@ -47,32 +47,13 @@ public String getUser()
         StringBuilder sb = new StringBuilder("ConnectorIdentity{");
         sb.append("user='").append(user).append('\'');
         principal.ifPresent(principal -> sb.append(", principal=").append(principal));
+        role.ifPresent(role -> sb.append(", role=").append(role));
```
Separate this commit into:
- `Remove ConnectorIdentity#equals`
- `Improve ConnectorIdentity#toString`

`toString` is a minor change, but change to `equals` is not and should not be hidden in a innocently-looking "refactor ..." commit

findepi commented:
Just a thought -- i think it would be safer not just to remove `ConnectorIdentity#equals`. You could leave the method with as this:
```
@Override
public boolean equals(Object o)
{
    // TODO remove equals completely after a few months
    throw new UnsupportedOperationException();
}
```

luohao commented:
> Just a thought -- i think it would be safer not just to remove `ConnectorIdentity#equals`. You could leave the method with as this:
> 
> ```
> @Override
> public boolean equals(Object o)
> {
>     // TODO remove equals completely after a few months
>     throw new UnsupportedOperationException();
> }
> 
> @Override
> public int hashCode()
> {
>     // TODO remove equals completely after a few months
>     // hashCode() needs to be overridden too, since j.u.HashMap may skip equals() check when `==`.
>     throw new UnsupportedOperationException();
> }
> ```

This sounds like a better strategy. I updated the diff and split it into two commits.

luohao commented:
@findepi Updated the commit message.
